Core/src/org/sleuthkit/autopsy/events/AutopsyEventPublisher.java

```diff
@@ -83,29 +120,22 @@ synchronized public void removeSubscriber(Collection<String> eventNames, Autopsy
      * @param eventNames The event the subscriber is no longer interested in.
      * @param subscriber The subscriber to remove.
      */
-    synchronized public void removeSubscriber(String eventName, AutopsyEventSubscriber subscriber) {
-        Set<AutopsyEventSubscriber> subscribers = subscribersByEvent.getOrDefault(eventName, null);
-        if (null != subscribers) {
-            subscribers.remove(subscriber);
-        }
+    synchronized public void removeSubscriber(String eventName, PropertyChangeListener subscriber) {
+        localPublisher.removeSubscriber(eventName, subscriber);
     }
 
     /**
-     * Publishes an event to all registered subscribers, even if a subscriber
-     * throws an exception.
+     * Publishes an event.
      *
-     * @param event The event to be published.
+     * @param event The event to publish.
+     * @throws JMSException
      */
-    synchronized public void publish(AutopsyEvent event) {
-        Set<AutopsyEventSubscriber> subscribers = subscribersByEvent.getOrDefault(event.getPropertyName(), null);
-        if (null != subscribers) {
-            for (AutopsyEventSubscriber subscriber : subscribers) {
-                try {
-                    subscriber.receiveEvent(event);
-                } catch (Exception ex) {
-                    logger.log(Level.SEVERE, "Exception thrown by subscriber", ex);
-                }
-            }
+    synchronized public void publish(AutopsyEvent event) throws JMSException {
+        event.setSourceType(AutopsyEvent.SourceType.LOCAL);
+        localPublisher.publish(event);
+        if (null != remotePublisher) {
```

Comment:
I'm not sure what you mean by "publish it back locally."  When an object, say the IngestManager, publishes an event (fires a property change) it needs to be sent to the subscribers on the local or source node, as well as to the subscribers on other nodes, the remote nodes. The LocalEventPublisher (an improved version of PropertyChangeSupport) handles the publication of events to subscribers on the local or source node via calls to propertyChange(), and the RemoteEventPublisher handles the publication of outgoing events to the remote nodes by sending JMS messages. This is what happens in AutopsyEventPublisher.publish(). When a node receives a JMS event message, it needs to be published to the subscribers on the local node. That's why the RemoteEventPublisher gets a reference to the LocalEventPublisher which, like PropertyChangeSupport, encapsulates the local subscribers. The action happens in RemoteEventPublisher.MessageReceiver.onMessage() in a JMS thread. If I have not answered your question, please rephrase it.